src/modules/ModuleTradeShared/composable.pair-input.ts

```diff
@@ -24,6 +25,42 @@ export function useLocalStorageAddrsOrigin(key: string, isActive?: Ref<boolean>)
   return offable
 }
 
+export function useRouteAddrsOrigin({
+  baseToken,
+  additionalBaseToken,
+  isActive,
+}: {
+  baseToken?: Address
+  additionalBaseToken?: Address
+  isActive?: Ref<boolean>
+}) {
+  const params = {
+    tokenA: useRouteParams('tokenA'),
+    tokenB: useRouteParams('tokenB'),
+  }
+
+  return computed({
+    get: () => {
+      if (!isActive?.value) return emptyPair()
+
+      const tokenA =
+        (typeof params.tokenA.value === 'string' && isAddress(params.tokenA.value) && params.tokenA.value) || null
+      const tokenB =
+        (typeof params.tokenB.value === 'string' && isAddress(params.tokenB.value) && params.tokenB.value) || null
```

Comment:
```suggestion
      const addressOrNull = (x: unknown) => isAddress(x) ? x : null
      const tokens = buildPair((type) => addressOrNull(params[type].value))
```

Less code, no duplication and still type-safe.